vars/fortifyStage.groovy

```diff
@@ -0,0 +1,126 @@
+def call(body) {
+  echo "called fortify stage"
+  def config = [:]
+  body.resolveStrategy = Closure.DELEGATE_FIRST
+  body.delegate = config
+  body()
+
+  if (config.directory == null) {
+      config.directory = '.'
+  }
+
+  if (config.projname == null) {
+      config.projname = "${env.JOB_BASE_NAME}"
+  }
+
+  if(config.notifyMessages == null) {
+    def notifyMessages = []
+  }
+
+  if(config.artifactId == null) {
+    artifactId = readMavenPom().getArtifactId()
+  }
+
+  if(config.version == null) {
+    version = readMavenPom().getVersion()
+  }
+
+  if(config.applyGates == null) {
+    applyGates = true
+  }
+
+  if(config.failOnGates == null) {
+    failOnGates = true
+  }
+
+  node('fortify-sca') {
+    echo "in fortify node"
+    stage ('Debug'){
+      sh "which mvn"
+      sh "which ant"
+      sh "ant -version"
+      sh "mvn -v"
+    }
+
+    def tmpDir = pwd(tmp: true)
+
+    if (config.mavenSettings == null) {
+        config.mavenSettings = "${tmpDir}/settings.xml"
+        stage('Configure Maven') {
+            def mavenSettings = libraryResource 'gov/va/maven/settings.xml'
+            writeFile file: config.mavenSettings, text: mavenSettings
+        }
+    }
+
+    // recheck out scm in case the stage before changed any poms (like a release, for instance)
+    stage('Checkout SCM') {
+        checkout scm
+    }
+
+    stage ('Fortify'){
+        lock(resource: "lock_fortify_${env.NODE_NAME}_${artifactId}") {
```

Comment:
Can there not be simultaneous fortify scans being run?

Reply:
Lets remove this resource lock as I don't think we need it. We can control concurrent scans by setting the number of executors.